aurelia-slickgrid/src/aurelia-slickgrid/services/sort.service.ts

```diff
@@ -181,20 +192,29 @@ export class SortService {
 
   dispose() {
     // unsubscribe local event
-    if (this._subscriber && typeof this._subscriber.unsubscribe === 'function') {
-      this._subscriber.unsubscribe();
+    if (this._slickSubscriber && typeof this._slickSubscriber.unsubscribe === 'function') {
+      this._slickSubscriber.unsubscribe();
     }
 
     // unsubscribe all SlickGrid events
     this._eventHandler.unsubscribeAll();
   }
 
   /**
-   * A simple function that is attached to the subscriber and emit a change when the sort is called.
+   * A simple function that will be called to emit a change when a sort changes.
    * Other services, like Pagination, can then subscribe to it.
    * @param sender
    */
-  emitSortChangedBy(sender: string) {
-    this._subscriber.subscribe(() => this.onSortChanged.publish('sortService:changed', `onSortChanged by ${sender}`));
+  emitSortChanged(sender: 'local' | 'remote') {
```

Comment:
same comment as in the filter service where the logic may be able to be simplified if only two strings are allowed